src/Umbraco.Web.UI.Client/src/common/services/tinymce.service.js

```diff
@@ -338,15 +338,28 @@ function tinyMceService($rootScope, $q, imageHelper, $locale, $http, $timeout, s
                     body_class: "umb-rte",
 
                     //see http://archive.tinymce.com/wiki.php/Configuration:cache_suffix
-                    cache_suffix: "?umb__rnd=" + Umbraco.Sys.ServerVariables.application.cacheBuster,
+                    cache_suffix: "?umb__rnd=" + Umbraco.Sys.ServerVariables.application.cacheBuster
+                };
+
+                // Need to check if we are allowed to UPLOAD images
+                // This is done by checking if the insert image toolbar button is available
+                var insertMediaButtonFound = false;
+                args.toolbar.forEach(toolbarItem => {
```

Comment:
This seems like duplicate code, we should have a `function isMediaPickerEnabled` so we can re-use this instead of doing this foreach loop and boolean check in more than one place.